code/osapi/dialogs.cpp

```diff
@@ -230,7 +230,7 @@ namespace os
 			boxData.flags = SDL_MESSAGEBOX_ERROR;
 			boxData.message = boxText.c_str();
 			boxData.title = "Error!";
-			boxData.window = os::getSDLMainWindow();
+			boxData.window = !(Cmdline_fullscreen_window || Cmdline_window) ? NULL : os::getSDLMainWindow();
```

Comment:
Since you duplicated this code for every invocation, please move it to a common function